### PR TITLE
fix(Menu): Menu component improvements

### PR DESCRIPTION
**Pull request by sergivillar**
No visual changes, only fixing a bug.
Attached a video of the issue found (menu usage with changing render, autocomplete component)

https://user-images.githubusercontent.com/6709771/149767985-43a5fde0-4d6e-4b6d-a242-f406a39cc994.mp4

This componente can be tested in Menu component and also in Custom field example (both in storybook)

### CODE REVIEW

File: src/menu.tsx [hunk 1]
```diff
@@ -65,39 +66,38 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width, position =
...
+                    top: 'auto',
+                    maxHeight: Math.min(topTarget, MAX_HEIGHT_DEFAULT),
                     transformOrigin: 'center bottom',
                 });
             } else {
                 setItemsComputedProps({
-                    width: widthTarget,
-                    top,
-                    left: leftDirection,
+                    top: '100%',
```
Comment (sergivillar) [inline src/menu.tsx:88]: moved to percentage values, this way we should not think about absolute px values.

File: src/menu.tsx [hunk 2]
```diff
@@ -162,20 +162,21 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width, position =
...
+                    style={{position: 'fixed', top: 0, bottom: 0, left: 0, right: 0}}
+                    onClick={(e) => {
                         cancelEvent(e);
+                        setIsMenuOpen(false);
                     }}
-                    disableScroll
-                >
-                    {renderMenu(menuProps)}
-                </Overlay>
+                />
```
[the remark this answers is not in the document]
Comment (pladaria) [reply src/menu.tsx:175]: what makes this div different to the previous implementation?
Overlay has the same styles:
```
const defaultStyle: React.CSSProperties = {
    position: 'fixed',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    zIndex: 12,
    outline: 'none',
    WebkitTapHighlightColor: 'transparent',
};
```
On the other hand, Overlay has logic to cancel the scroll. IIRC, this is needed in iOS devices, where you don't want to scroll the bottom layers when the overlay is displayed.

[the remark this answers is not in the document]
Comment (sergivillar) [reply src/menu.tsx]: right,changed to use overlay again, but without renderMenu as a children

**Review by atabel — APPROVED** [review]

File: src/menu.tsx [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
-import Overlay from './overlay';
 import {createUseStyles} from './jss';
 import {cancelEvent} from './utils/dom';
+import Overlay from './overlay';
 
 const MAX_HEIGHT_DEFAULT = 416;
 
 const useStyles = createUseStyles(({colors}) => ({
     menuContainer: {
+        zIndex: 12,
```
[the remark this answers is not in the document]
Comment (sergivillar) [reply src/menu.tsx:11]: copied from https://github.com/Telefonica/mistica-web/blob/master/src/overlay.tsx

previously Menu was render inside Overlay and it was shown above all page content, now that it lives outside that, the behaviour must be mimic

**Review by pladaria — APPROVED** [review]

**Pull request merged**